tests/test_mask_parameter.py

```diff
@@ -0,0 +1,111 @@
+        self.toas = get_TOAs("J1713+0747_NANOGrav_11yv0_short.tim")
+
+    def test_mjd_mask(self):
+        mp = maskParameter("test1", key="mjd", key_value=[54000, 54100])
+        assert mp.key == "mjd"
+        assert mp.key_value == [54000, 54100]
+        assert mp.value == None
+        select_toas = mp.select_toa_mask(self.toas)
+        assert np.all(self.toas.table["mjd_float"][select_toas] <= 54100)
+        assert np.all(self.toas.table["mjd_float"][select_toas] >= 54000)
+        mp_str_keyval = maskParameter("test2", key="mjd", key_value=["54000", "54100"])
+        assert mp_str_keyval.key_value == [54000, 54100]
+        mp_value_switch = maskParameter("test1", key="mjd", key_value=[54100, 54000])
+        assert mp_value_switch.key_value == [54000, 54100]
+        with pytest.raises(ValueError):
+            mp_str_keyval = maskParameter("test2", key="mjd", key_value=["54000"])
+
+    def test_freq_mask(self):
+        mp = maskParameter("test2", key="freq", key_value=[1400, 1430])
+        assert mp.key_value == [1400 * u.MHz, 1430 * u.MHz]
+        select_toas = mp.select_toa_mask(self.toas)
+        assert np.all(self.toas.table["freq"][select_toas] <= 1430 * u.MHz)
+        assert np.all(self.toas.table["freq"][select_toas] >= 1400 * u.MHz)
+        mp_str = maskParameter("test2", key="freq", key_value=["1400", "2000"])
+        assert mp_str.key_value == [1400 * u.MHz, 2000 * u.MHz]
+        mp_switch = maskParameter("test2", key="freq", key_value=[2000, 1400])
+        assert mp_switch.key_value == [1400 * u.MHz, 2000 * u.MHz]
+        mp_quantity = maskParameter(
+            "test2", key="freq", key_value=[2000 * u.MHz, 1400 * u.MHz]
+        )
+        assert mp_quantity.key_value == [1400 * u.MHz, 2000 * u.MHz]
+        with pytest.raises(ValueError):
+            mp_wrong_keyvalue = maskParameter("test2", key="freq", key_value=[1400])
+
+    def test_tel_mask(self):
+        mp_ao = maskParameter("test2", key="tel", key_value="ao")
+        assert mp_ao.key_value == ["arecibo"]
+        select_toas = mp_ao.select_toa_mask(self.toas)
+        print(self.toas.table["obs"][select_toas])
+        assert np.all(self.toas.table["obs"][select_toas] == "arecibo")
+        mp_gbt = maskParameter("test2", key="tel", key_value=["gbt"])
+        assert mp_gbt.key_value == ["gbt"]
+        select_toas = mp_gbt.select_toa_mask(self.toas)
+        assert np.all(self.toas.table["obs"][select_toas] == "gbt")
+        mp_special_obs1 = maskParameter("test2", key="tel", key_value="@")
+        assert mp_special_obs1.key_value == ["barycenter"]
+        mp_special_obs2 = maskParameter("test2", key="tel", key_value=0)
+        assert mp_special_obs2.key_value == ["geocenter"]
+        with pytest.raises(ValueError):
+            mp_wrong_keyvalue = maskParameter(
+                "test2", key="tel", key_value=["gbt", "ao"]
+            )
+
+    def test_name_mask(self):
+        # Not sure about the use case for name mask
```

Comment:
`name` is often the input file the TOAs were generated from. One could reasonably select all the TOAs that were generated from a single file.

Reply:
Ok, got it, can make some more test on it.

Reply:
Test added.